client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -372,12 +372,14 @@ class CalypsoifyIframe extends Component<
 		}
 
 		if ( EditorActions.GetGutenboardingStatus === action ) {
-			const isGutenboarding = this.props.siteCreationFlow === 'gutenboarding' && ! this.props.plan;
+			const isGutenboarding =
+				this.props.siteCreationFlow === 'gutenboarding' &&
+				( ! this.props.plan || this.props.plan.is_free ); // prevent showing StepByStepLaunch on sites with paid plans
```

Comment:
Putting the `( ! this.props.plan || this.props.plan.is_free )` check here may cause some confusion, since the `isGutenboarding` variable now carries a different meaning that is not just to do with the site creation flow.

Would it be better move the check under a separate `siteHasFreePlan` variable, and pass that variable to `calypsoifyGutenberg`?
The code on the Editing Toolkit side could then check for `isGutenboarding && siteHasFreePlan`, making the code more explicit on what's going on

Reply:
I agree there is a better way to do this and also prevent confusion. Opened https://github.com/Automattic/wp-calypso/issues/47504 to unblock current PR so we can quickly fix the issue from the Calypso side and then implement a more solid solution that involves deploying ETK and WBE.

Reply:
Agreed. Thank you for opening a separate issue to track this.